evm/src/cpu/jumps.rs

```diff
@@ -0,0 +1,334 @@
+use once_cell::sync::Lazy;
+use plonky2::field::extension::Extendable;
+use plonky2::field::packed::PackedField;
+use plonky2::field::types::Field;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::ext_target::ExtensionTarget;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::cpu::columns::CpuColumnsView;
+use crate::cpu::kernel::aggregator::KERNEL;
+
+static INVALID_DST_HANDLER_ADDR: Lazy<usize> =
+    Lazy::new(|| KERNEL.global_labels["handle_invalid_jump_dst"]);
+
+pub fn eval_packed_exit_kernel<P: PackedField>(
+    lv: &CpuColumnsView<P>,
+    nv: &CpuColumnsView<P>,
+    yield_constr: &mut ConstraintConsumer<P>,
+) {
+    let jumps_lv = lv.general.jumps();
+
+    yield_constr.constraint_transition(
```

Comment:
Just to double check, with this design, we can use `PUSH 0` + `EXIT_KERNEL` to begin executing a user contract right? Looks like this code would read `is_kernel_mode=0` as we want.

And I guess if we wanted a system call to call another system call, we could do `PUSH return_label` + `%add_const(0x100000000)` + `%jump(inner_syscall)`, so that `inner_syscall`'s `EXIT_KERNEL` will see `is_kernel_mode=1`? (I don't think it seems like an essential feature, but if we did want to do it.)